runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowPipelineTranslator.java

```diff
@@ -656,7 +656,12 @@ private long addOutput(PValue value, Coder<?> valueCoder) {
 
       Map<String, Object> outputInfo = new HashMap<>();
       addString(outputInfo, PropertyNames.OUTPUT_NAME, Long.toString(id));
-      addString(outputInfo, PropertyNames.USER_NAME, value.getName());
+
+      String stepName = getString(properties, PropertyNames.USER_NAME);
+      String generatedName = String.format(
+          "%s.out%d", stepName, outputInfoList.size());
```

Comment:
The tags themselves also have to be ignored - the unit tests cover both TupleTags with specified strings and TupleTags without specified strings.